tests/setup_nodes.py

```diff
@@ -636,11 +653,11 @@ async def setup_full_system(
                 node1_port,
                 rpc1_port,
                 b_tools,
+                connect_to_daemon_port,
```

Comment:
it's not clear to me why you need to pass in this port here. You're already passing in `b_tools`, which knows which port the daemon runs on. You could just ask it.

Unless there's some shenanigans where you're actually connecting to some *other* BlockTool's daemon.. but in that case, I would think there would be some comments in the code making the reader aware of that.

Reply:
That's true. I shouldn't need to both pass the port AND have it in the config. I did this to make it explicit to the caller that it needs to be set

Reply:
one of my changes (that weren't in when you started this patch, but has landed since) made all BlockTools instances always pick a free port and configure the daemon with it. So, in `main` it should always be set.